controllers/builder/kogitoserverlessbuild_manager_test.go

```diff
@@ -0,0 +1,185 @@
+	assert.NotNil(t, build)
+	assert.Equal(t, 1, len(build.Spec.BuildArgs))
+	assertContainsPersistence(t, build.Spec.BuildArgs, 0)
+	restoreControllersConfig(t)
+}
+
+func prepareGetOrCreateBuildTest(t *testing.T, currentPlatform *operatorapi.SonataFlowPlatform) sonataFlowBuildManager {
+	initializeControllersConfig(t)
+	platforms := operatorapi.NewSonataFlowPlatformList()
+	platforms.Items = []operatorapi.SonataFlowPlatform{*currentPlatform}
+	cli := test.NewSonataFlowClientBuilder().WithRuntimeObjects(&platforms).Build()
+	buildManager := sonataFlowBuildManager{
+		client: cli,
+	}
+	return buildManager
+}
+
+func Test_addPersistenceExtensionsWithEmptyArgs(t *testing.T) {
+	initializeControllersConfig(t)
+	buildTemplate := &operatorapi.BuildTemplate{}
+	addPersistenceExtensions(buildTemplate)
+	assert.Equal(t, 1, len(buildTemplate.BuildArgs))
+	assertContainsPersistence(t, buildTemplate.BuildArgs, 0)
+	restoreControllersConfig(t)
+}
+
+func Test_addPersistenceExtensionsWithNoQuarkusExtensionsArg(t *testing.T) {
+	initializeControllersConfig(t)
+	buildTemplate := &operatorapi.BuildTemplate{
+		BuildArgs: []v1.EnvVar{
+			{Name: "VAR1"},
+		},
+	}
+	addPersistenceExtensions(buildTemplate)
+	assert.Equal(t, 2, len(buildTemplate.BuildArgs))
+	assertContainsPersistence(t, buildTemplate.BuildArgs, 1)
+	restoreControllersConfig(t)
+}
+
+func Test_addPersistenceExtensionsWithQuarkusExtensionsArgAndNoPersistenceExtensions(t *testing.T) {
+	initializeControllersConfig(t)
+	buildTemplate := &operatorapi.BuildTemplate{
+		BuildArgs: []v1.EnvVar{
+			{Name: "VAR1"},
+			{Name: "QUARKUS_EXTENSIONS", Value: "org.acme:org.acme.library:1.0.0"},
+		},
+	}
+	addPersistenceExtensions(buildTemplate)
+	assert.Equal(t, 2, len(buildTemplate.BuildArgs))
+	assertContainsPersistence(t, buildTemplate.BuildArgs, 1)
+	restoreControllersConfig(t)
+}
+
+func Test_addPersistenceExtensionsWithQuarkusExtensionsArgAndPersistenceExtensions(t *testing.T) {
+	initializeControllersConfig(t)
+	buildTemplate := &operatorapi.BuildTemplate{
+		BuildArgs: []v1.EnvVar{
+			{Name: "VAR1", Value: "VALUE1"},
+			{Name: "QUARKUS_EXTENSIONS", Value: "org.acme:org.acme.library:1.0.0,io.quarkus:quarkus-jdbc-postgresql:8.8.0.Final"},
+		},
+	}
+	addPersistenceExtensions(buildTemplate)
+	assert.Equal(t, 2, len(buildTemplate.BuildArgs))
+	assert.Equal(t, v1.EnvVar{Name: "VAR1", Value: "VALUE1"}, buildTemplate.BuildArgs[0])
+	assert.Equal(t, v1.EnvVar{Name: "QUARKUS_EXTENSIONS", Value: "org.acme:org.acme.library:1.0.0,io.quarkus:quarkus-jdbc-postgresql:8.8.0.Final"}, buildTemplate.BuildArgs[1])
+	restoreControllersConfig(t)
+}
+
+func initializeControllersConfig(t *testing.T) {
+	// emulate the controllers config initialization
+	cfg, err := cfg.InitializeControllersCfgAt("../cfg/testdata/controllers-cfg-test.yaml")
+	assert.NoError(t, err)
+	assert.NotNil(t, cfg)
+	assert.Equal(t, 3, len(cfg.PostgreSQLPersistenceExtensions))
+}
+
+func restoreControllersConfig(t *testing.T) {
```

Comment:
Mind moving this function to a global util test file? I think I had something like this when I first implemented the configuration feature. So we keep it in one place to reuse across the code base.